modular_skyrat/modules/apc_arcing/apc.dm

```diff
@@ -0,0 +1,55 @@
+/obj/machinery/power/apc
+	/// Has the APC been protected against arcing?
+	var/arc_shielded = FALSE
+
+/obj/machinery/power/apc/examine()
+	. = ..()
+	. += "It [arc_shielded ? "has" : "does not have"] arc shielding installed."
+	if(panel_open)
+		if(arc_shielded)
+			. += "The arc shielding could be removed with a <b>wrench</b>."
+		else
+			. += "It could be arc shielded with a <b>sheet of bronze</b>."
+
+/obj/machinery/power/apc/process()
+	. = ..()
+	var/excess = surplus()
+	if(!cell || shorted)
+		return
+	if(!(excess >= APC_ARC_LOWERLIMIT) || arc_shielded)
+		return
+	var/shock_chance = 5
+	if(excess >= APC_ARC_UPPERLIMIT)
+		shock_chance = 15
+	else if(excess >= APC_ARC_MEDIUMLIMIT)
+		shock_chance = 10
+	if(prob(shock_chance))
+		var/list/shock_mobs = list()
+		for(var/creature in view(get_turf(src), 5)) // We only want to shock a single random mob in range, not all.
+			if(isliving(creature))
+				shock_mobs += creature
+		if(shock_mobs.len)
+			var/mob/living/living_target = pick(shock_mobs)
```

Comment:
```suggestion
	if(prob(shock_chance))
		var/list/shock_mobs = list()
		for(var/mob/living/creature in viewers(get_turf(src), 5)) // We only want to shock a single random mob in range, not all.
			shock_mobs += creature
		if(length(shock_mobs))
			var/mob/living/living_target = pick(shock_mobs)
```

Reply:
Won't `viewers()` mean the blind are shock immune? That doesn't make sense.

Reply:
> Won't `viewers()` mean the blind are arc-immune? That doesn't make sense.

byond's concept of blindness isnt what ss13 uses for blind users. blindness is for BYOND darkness, aka byond-FOV. You can also use hearers() to ignore this.